nix/test_devshell.py

```diff
@@ -0,0 +1,84 @@
+import os
+import shutil
+import subprocess
+import pytest
+
+
+ALL_LCS = ["openssl-3.0", "openssl-1.1.1",
+           "openssl-1.0.2", "libressl", "aws-lc"]
+
+
+def translate_lc(lc: str):
+    # awslc for S2N_LIBCRYPTO is special.
+    if "awslc" in lc:
+        return "aws-lc"
+    else:
+        return lc
+
+
+@pytest.mark.parametrize("lc", [os.getenv("S2N_LIBCRYPTO")])
+def test_s2n_libcrypto(lc):
+    # Validate S2N_LIBCRYPTO is in the CMAKE_INCLUDE_PATH
+    assert lc is not None
+    libcrypto = translate_lc(lc)
+    include_path = os.getenv("CMAKE_INCLUDE_PATH")
+    assert include_path is not None
+    assert libcrypto in ALL_LCS
+    assert libcrypto in include_path
+
+
+@pytest.mark.parametrize("lc", [os.getenv("S2N_LIBCRYPTO")])
+def test_s2n_libcrypto_uniq(lc):
+    # Make certain we only have the preferred libcrypto in CMAKE_INCLUDE_PATH.
+    assert lc is not None
+    libcrypto = translate_lc(lc)
+    include_path = os.getenv("CMAKE_INCLUDE_PATH")
+    assert include_path is not None
+    ALL_LCS.remove(libcrypto)
+    for library in ALL_LCS:
```

Comment:
Maybe weird question, it's been a while since I touched python: Since you're modifying a global variable, does that affect other tests using that global variable? Should you be operating on a copy of ALL_LCS instead?

Reply:
updated.